arch/risc-v/src/common/riscv_internal.h

```diff
@@ -134,6 +136,59 @@
 #define getreg64(a)         (*(volatile uint64_t *)(a))
 #define putreg64(v,a)       (*(volatile uint64_t *)(a) = (v))
 
+#else
+/* Use ASM for rv64ilp32. This may be the unified way to go. */
```

Comment:
but, why not use inline assembler for non CONFIG_ARCH_RV64ILP32 too?

Reply:
It was to minimize any side effects. We sure can try that with the help of CI. Let me send another push later.